### DISCUSSION TITLE
Which virtuoso component do you use? And which one are you missing?

### DISCUSSION THREAD

**Discussion by petyosi** [petyosi/react-virtuoso#695] [Polls]
Over the years, the library became a "family" of several similar components, each with its own distinct use cases. Which one do you use? This helps me understand and consider the focus of the library. Also, share your ideas and suggestions about components in the same space.

**Comment by williamstein** [comment]
So far I only use the flat list.  The one interesting functionality I had to add is a way to preserve and restore the scroll state, which is a little more difficult than it could be, given the current API.   Here's the code I wrote:  https://github.com/sagemathinc/cocalc/blob/master/src/packages/frontend/components/virtuoso-scroll-hook.ts

One issue I hit is with rendering codemirror editors, where their size changes in funny ways right when virtuoso is measuring the size.  I had to put in some custom size code to handle this edge case, but it might be revealing an issue.   Another issue I hit was trying to use Virtuoso inside of css transforms -- fortunately it was just a simple scale transform, so I moved it inside my row component and used custom sizing and that solved my problem as well.   I guess the point is that I pushed virtuoso pretty hard in my application, and was always able to find a robust solution to my problems using the existing API, which is a good sign!    After using Virtuoso in many ways across cocalc.com, I definitely really like it, and will continue donating every month!
